### DISCUSSION TITLE
Noob question about the buttons wiring

### DISCUSSION THREAD

**Discussion by acehog** [lcintron/star-wars-ge-droid-pod#2] [Q&A] [answered]
What does this mean:

"Wire or solder the tactile push buttons to pins 15 and 20 (& GND). These will be pulled-up."
Install normally open button between pin 15 and ground and the other between pin 20 and ground?
And how to pull them up?
Thanks!

**Comment by mkanoap** [comment, marked as answer]
He's saying that the software already configures those pins to pull up.  Meaning that they read as 1 when unconnected, and zero when connected to ground.  You don't have to do anything for that.

Pins 15 and 20 refer to GPIO pins, not the pins when counting on the header.   It's confusing, but this site is helpful.  https://pinout.xyz/

Both pins are conveniently 2 pins away from a ground pin, which allows the leads of common push buttons to span them directly.  I used something from this set.  https://www.amazon.com/dp/B07CMZCQS5?psc=1&ref=ppx_yo2ov_dt_b_product_details

Normally open means that the button does NOT connect the leads until the button is pushed.   When it is pushed, current can flow between them.  Which means if you connect a button between pin 15 and ground, when the button is pushed there will be a connection between those pins, and 15 will go from high (1) from being pulled up, to low (0) because it is connected to ground.  When you stop pressing the button, the connection is broken and the pin will read high again.

Looking at the pinout diagram referenced above, you will see that GPIO pin 15 and the ground pin near it are pins 10 and 6, or 10 and 14, depending how close to the edge you want it.  6 and 14 are both ground pins.  GPIO 20 and ground are pins 38 and 34 respectively.   Why do they have different numbers in the names?   It's not great, but the GPIO numbers refer to pins on chips on the pi, not the location on the header.  Different devices with the same chips might break those GPIO pins out to different places.   By referring to the GPIO pins in software instead of hardware numbers, the same software can be portable to different devices.  

So you connect a button between pin 10 (GPIO 15) and 6 (ground), and another button between 38 (GPIO 20) and 34 (ground).   If you are not connecting the buttons directly, but are using wires, you can pick any pin labeled ground.   In fact you could use the same ground pin for both buttons.

edit: moved misplaced comma that completely changed the meaning of the sentance.

**Comment by acehog** [comment]
Thank you very much for the clear explanation!
Helped me and hopefully others.
Struggling getting the code to work still. Great educational experience for me!
Alan
